django/university/controllers/ClassController.py

```diff
@@ -1,24 +1,29 @@
-from university.models import Class, SlotClass, Slot, SlotProfessor
+from university.models import Class, Professor, Slot, SlotProfessor
 
 
 class ClassController:
     @staticmethod
-    def get_professors(slot_obj):
-        try:
-            professor = slot_obj.slotprofessor.professor
-            professors = [
-                {'id': professor.id, 'acronym': professor.professor_acronym, 'name': professor.professor_name}]
-        except SlotProfessor.DoesNotExist:
-            professors = []
+    def get_professors(slot):
+        slot_professors = SlotProfessor.objects.filter(slot_id=slot.id).values()
+        
+        professors = []
+        
+        for slot_professor in slot_professors:
+            professor = Professor.objects.get(id=slot_professor['professor_id'])
+            professors.append({
+                'id': professor.id,
+                'acronym': professor.professor_acronym,
+                'name': professor.professor_name
```

Comment:
Do we have the professor_link to append here? Or in the slotInfo (professors link)

Reply:
The information is not present in the schema of the backend. The scrapper will have to be changed to insert that information, unfortunately